File: Q_771.cpp
class Solution {
public:
    int numJewelsInStones(string J, string S) {
        set<char> myset;
        int i,count=0;
        for(i=0;i<J.size();i++)
            myset.insert(J[i]);
        for(i=0;i<S.size();i++)
            if(myset.find(S[i])!=myset.end())
                count++;
        return count;
    }
};

File: python/Q_771.py
import hashlib
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        count=0
        mymap={}
        for c in J:
            mymap[c]=c
        for c in S:
            if c in mymap
                count=count+1
        return count
